openlibrary/plugins/openlibrary/lists.py

```diff
@@ -28,6 +29,90 @@ def GET(self):
         return render_template("lists/home")
 
 
+@public
+def get_seed_info(doc):
+    """Takes a thiing, determines what type it is, and returns a seed summary"""
+    if doc.key.startswith("/subjects/"):
+        seed = doc.key.split("/")[-1]
+        if seed.split(":")[0] not in ("place", "person", "time"):
+            seed = f"subject:{seed}"
+        seed = seed.replace(",", "_").replace("__", "_")
+        seed_type = "subject"
+        title = doc.name
+    else:
+        seed = {"key": doc.key}
+        if doc.key.startswith("/authors/"):
+            seed_type = "author"
+            title = doc.get('name', 'name missing')
+        elif doc.key.startswith("/works"):
+            seed_type = "work"
+            title = doc.get("title", "untitled")
+        else:
+            seed_type = "edition"
+            title = doc.get("title", "untitled")
+    return {
+        "seed": seed,
+        "type": seed_type,
+        "title": web.websafe(title),
```

Comment:
Be careful with this; it's generally the case that websafing happens in the template file, so this might get double escaped.